akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -475,6 +474,21 @@ private[akka] class RemoteActorRefProvider(
 
   def getDefaultAddress: Address = transport.defaultAddress
 
+  // no need for volatile, only intended as cached value, not necessarily a singleton value
+  private var serializationInformationCache: OptionVal[Serialization.Information] = OptionVal.None
+  @InternalApi override private[akka] def serializationInformation: Serialization.Information =
+    serializationInformationCache match {
+      case OptionVal.Some(info) ⇒ info
+      case OptionVal.None ⇒
+        if ((transport eq null) || (transport.defaultAddress eq null))
+          local.serializationInformation // address not know yet
```

Comment:
When does this happen? Isn't the address always available from config if remoting is used? Also, won't this lead to the local serialization info potentially being put here and never replaced?

Reply:
it's the dreaded startup case, if accessed before completely initialized

The address is not known until bound, port 0.